src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -135,11 +135,14 @@ internal static void EnsureNullableAttributeExists(ImmutableArray<ParameterSymbo
         {
             foreach (var parameter in parameters)
             {
-                if (parameter.Type.ContainsNullableReferenceTypes())
+                var location = parameter.GetNonNullSyntaxNode().Location;
+                var parameterType = parameter.Type;
+                parameterType.ReportAnnotatedUnconstrainedTypeParameterIfAny(location, diagnostics);
```

Comment:
>ReportAnnotatedUnconstrainedTypeParameterIfAny [](start = 30, length = 46)

I understand the appeal of bundling this check here (paired with check for existence of Nullable attribute, as in other places). Maybe the method should be renamed to reflect that.
Maybe `CheckParametersForNullability`? (or some better name)